clang/test/SemaSYCL/intel-max-work-group-size.cpp

```diff
@@ -30,6 +30,14 @@ class Functor {
   [[intel::max_work_group_size(16, 16, 16)]] [[intel::max_work_group_size(32, 32, 32)]] void operator()(int) const; // expected-warning {{attribute 'max_work_group_size' is already applied with different arguments}} expected-note {{previous attribute is here}}
 };
 
+class FunctorC {
+public:
+  [[sycl::reqd_work_group_size(64, 64, 64)]] // expected-note {{conflicting attribute is here}}
+  [[intel::max_work_group_size(16, 16, 16)]] void operator()() const; //expected-error{{'max_work_group_size' attribute conflicts with 'reqd_work_group_size' attribute}}
+  [[sycl::reqd_work_group_size(64, 64, 64)]] // expected-note {{conflicting attribute is here}}
+  [[intel::max_work_group_size(16, 16, 16)]] void operator()(int) const; //expected-error{{'max_work_group_size' attribute conflicts with 'reqd_work_group_size' attribute}}
```

Comment:
Is this the same test as one above?

Reply:
They are slightly different in AST:
CXXMethodDecl 0x55efc07ffa98 <line:4:46, col:64> col:51 operator() 'void () const'

vs
CXXMethodDecl 0x55efc07ffe48 <line:6:46, col:67> col:51 operator() 'void (int) const'

Reply:
I'm not sure if I am missing something. Isn't that just because the method is defined with and without the parameter. I was wondering what the purpose of test is with respect to attribute testing.

Reply:
>>I believe this tests where diagnostic is not triggered (first method) vs diagnostic is triggered (second method) .

Correct.

>> I don't know if I'm missing something here

I do not think you are missing anything here. 

>>why would function signature affect attribute diagnostics

This should not affect.

Sorry i misunderstood your question. I will fix the tests.